cmd/snap-bootstrap/partition/encrypt.go

```diff
@@ -133,3 +161,61 @@ func cryptsetupClose(name string) error {
 	}
 	return nil
 }
+
+func cryptsetupAddKey(key EncryptionKey, rkey RecoveryKey, node string) error {
+	// create a named pipe to pass the recovery key
+	fpath := filepath.Join(dirs.SnapRunDir, "tmp-rkey")
+	if err := syscall.Mkfifo(fpath, 0600); err != nil {
+		return fmt.Errorf("cannot create named pipe: %v", err)
+	}
+	defer os.RemoveAll(fpath)
+
+	// add a new key to slot 1 reading the passphrase from the named pipe
+	args := []string{
+		// add a new key
+		"luksAddKey",
+		// the encrypted device
+		node,
+		// batch processing, no password verification
+		"-q",
+		// read existing key from stdin
+		"--key-file", "-",
+		// store it in keyslot 1
```

Comment:
It would be good to explain in the code comment why slot 1 is used and how that is significant.

Reply:
At this point it doesn't need to be specifically in keyslot 1 and since this is the first key to be added it would end up using keyslot 1 (but I think this ordering is not explicitly documented). The parameter is there to guarantee that we have a predictable slot number in case we decide to kill the other slots later. Will add a comment about that.

Reply:
Can you include that explanation in the code please.

Reply:
Note added.